### PR TITLE
Disable any package download plugin by default

### PR DESCRIPTION
**Pull request by PatoBeltran**
## Bug
Fixes: https://github.com/NuGet/Home/issues/7377

## Fix
This PR disables by default any package download plugin and adds an environment variable to enable them.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs [hunk 1]
```diff
@@ -131,7 +132,7 @@ public async Task<IEnumerable<PluginDiscoveryResult>> FindAvailablePluginsAsync(
             var pluginCreationResults = new List<PluginCreationResult>();
 
             // Fast path
-            if (source.PackageSource.IsHttp && IsPluginPossiblyAvailable())
+            if (source.PackageSource.IsHttp && IsPluginPossiblyAvailable() && AreDownloadPlugingsEnabled())
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs]: There could be multiple calls here, you'd ideally want the calculation of the AreDownloadPluginsEnabled to happen once.

Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs]: If we memoize it, can it be the case that throughout the life of the `PluginManager` the value of the environment variable changes and it's not reflected on this method?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs]: We already do that with the plugin paths. 
https://github.com/NuGet/NuGet.Client/blob/e0340ccda98518f3eccf92716ffe4f283625344a/src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs#L332-L337

**Comment by PatoBeltran** [comment]
@nkolev92 I added the reading of the env var to the `Initialize` method, the same way it's done for others.

**Review by nkolev92 — APPROVED** [review]
LGTM. 

I think we should wait on merging until we get closure in the internal thread.

File: test/NuGet.Clients.Tests/NuGet.Credentials.Test/PluginManagerMock.cs [hunk 2]
```diff
@@ -210,6 +210,9 @@ private void EnsureAllEnvironmentVariablesAreCalled(string pluginFilePath)
             _reader.Setup(x => x.GetEnvironmentVariable(
                     It.Is<string>(value => value == CredentialTestConstants.PluginHandshakeTimeoutEnvironmentVariable)))
                         .Returns("HandshakeTimeout");
+            _reader.Setup(x => x.GetEnvironmentVariable(
```
Comment (jainaashish) [inline test/NuGet.Clients.Tests/NuGet.Credentials.Test/PluginManagerMock.cs:213]: can you add a test where this env is disabled and test we dont initialize download plugin?

**Review by jainaashish — APPROVED** [review]
Add a test for negative case and then good.

**Comment by PatoBeltran** [comment]
Closing this in favor of https://github.com/NuGet/NuGet.Client/pull/2471
